Assets/Scripts/Player/WeaponInventory.cs

```diff
@@ -174,7 +203,7 @@ public void AddWeapon(WeaponData weapon)
 
     private void RemoveWeapon(int weaponIndex)
     {
-        if (state == State.NO_WEAPONS)
+        if (state == State.NO_WEAPONS || Weapons.Count == 1)

```

Comment:
If we are no longer allowing the player to drop all of their weapons then a lot of this code is redundant because of the assumptions I made when I wrote it initially.

Reply:
I can make it so the player also has a state for not holding a weapon, but that'll take time I don't have